### PR TITLE
Tech/wmt 787/make default contracted hours more generic

### PR DESCRIPTION
**Pull request by dconey646**

### CODE REVIEW

**Comment by caoimheb** [comment]
Unit tests for `calculate-contracted-hours`?

Also, there is a TODO in `tesst-calculate-available-points`. Do we have an answer for that yet?

**Comment by dconey646** [comment]
Unit tests have been added, forgot to commit them yesterday.

File: test/unit/points/test-calculate-contracted-hours.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  it('should return null when unsupported type is selected', function () {
+    offenderManagerTypeId = 4
+    /* eslint-disable no-unused-expressions */
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, offenderManagerTypeId)).to.be.null
+  })
+
+  it('should return contracted hours', function () {
+    offenderManagerTypeId = 1
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, offenderManagerTypeId)).to.be.an('object')
+  })
```
Comment (caoimheb) [inline test/unit/points/test-calculate-contracted-hours.js]: Can you add a few tests which check concrete values? For example, when passed an invalid contracted hours then the base is set to default, and one where base is consistent with contractedHoursPerWeek?

(Some of the tests from `test-calculate-available-points` might be better placed in this test class)

Comment (dconey646) [inline test/unit/points/test-calculate-contracted-hours.js]: Yea! Good shout, I will move most of them across then.

File: test/unit/points/test-calculate-contracted-hours.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  it(`returns default PSO contracted hours when offender manager type ID is ${OM_TYPE_IDS.PSO_C}`, function () {
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, OM_TYPE_IDS.PSO_C)).to.be.an('object')
+  })
+
+  it('returns other default contracted hours when offender manager type ID is an unexpected number', function () {
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, 1)).to.be.an('object')
+  })
+
+  it('uses defaultContractedHoursForBand when contractedHoursPerWeek is not provided', function () {
+    // TODO: check if zero for this is an acceptable answer or should it use defaultContractedHoursForBand too.
```
Comment (caoimheb) [inline test/unit/points/test-calculate-contracted-hours.js]: Would rather not merge a `TODO` into `master`. Could we add this to the questions spreadsheet maybe?

File: test/unit/points/test-calculate-contracted-hours.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  it(`returns default PSO contracted hours when offender manager type ID is ${OM_TYPE_IDS.PSO_B}`, function () {
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, OM_TYPE_IDS.PSO_B)).to.be.an('object')
+  })
+
+  it(`returns default PSO contracted hours when offender manager type ID is ${OM_TYPE_IDS.PSO_C}`, function () {
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, OM_TYPE_IDS.PSO_C)).to.be.an('object')
+  })
+
+  it('returns other default contracted hours when offender manager type ID is an unexpected number', function () {
+    expect(calculateContractedHours(contractedHoursPerWeek, defaultContractedHours, 1)).to.be.an('object')
```
Comment (caoimheb) [inline test/unit/points/test-calculate-contracted-hours.js]: Add expects for `baseHours` and `defaultContractedHoursForBand` in the returned object?

File: test/unit/points/test-calculate-contracted-hours.js (hunk 2 above)
Comment (dconey646) [inline test/unit/points/test-calculate-contracted-hours.js]: Yea, sure. I will fill that spreadsheet in now for it.

File: test/unit/points/test-calculate-contracted-hours.js (hunk 3 above)
Comment (dconey646) [inline test/unit/points/test-calculate-contracted-hours.js]: Yep, done now.

**Pull request merged**